### ISSUE TITLE
Advices about LOSS?

### ISSUE THREAD

**Issue by Joevaen** [MIC-DKFZ/nnUNet#338]
Hi, fabian!
Have you tried some other loss functions except DICE+CE? I am trying to improve nnunet in this way, may I get some advices?
Thank you!
Joevaen.

**Comment by FabianIsensee** [comment]
Hi,
I have tried a bunch of them and in my experience, Dice+CE works the best on most datasets. There may be some exceptions on single datasets and it may be worth trying that if you have a strong intuition.
There is also work by @JunMa11 whcih compares loss function on several datasets: https://arxiv.org/abs/2005.13449
Note that while they found an increase in performance with topk + dice over my current loss, I was unable to confirm that so far (I am using different/more datasets than they are but I am not optimizing hyperparameters (which I probably should))
Best,
Fabian

**Comment by JunMa11** [comment]
Hi @FabianIsensee , thank you very much for referring my paper:)

@Joevaen The loss odyssey paper is under major revision in MedIA, Here are some quantitative results about loss functions: https://github.com/JunMa11/SegLoss/tree/master/test

We evaluated and compared different loss functions on four datasets. The only optimized hyperparameter is the learning rate. We applied the grid search among 3e-3, 3e-4, 3e-5.

Best,
Jun

**Comment by Joevaen** [comment]
> Hi,
> I have tried a bunch of them and in my experience, Dice+CE works the best on most datasets. There may be some exceptions on single datasets and it may be worth trying that if you have a strong intuition.
> There is also work by @JunMa11 whcih compares loss function on several datasets: https://arxiv.org/abs/2005.13449
> Note that while they found an increase in performance with topk + dice over my current loss, I was unable to confirm that so far (I am using different/more datasets than they are but I am not optimizing hyperparameters (which I probably should))
> Best,
> Fabian

Thank you, fabian!

**Comment by Joevaen** [comment]
@JunMa11 @FabianIsensee 
A lot of thanks to you guys! :) 

> Hi @FabianIsensee , thank you very much for referring my paper:)
> 
> @Joevaen The loss odyssey paper is under major revision in MedIA, Here are some quantitative results about loss functions: https://github.com/JunMa11/SegLoss/tree/master/test
> 
> We evaluated and compared different loss functions on four datasets. The only optimized hyperparameter is the learning rate. We applied the grid search among 3e-3, 3e-4, 3e-5.
> 
> Best,
> Jun

:+1:

**Comment by FabianIsensee** [comment]
For each change in hyperparameters you need a new trainer class. 
Note that @JunMa11 did his experiments with nnUNetTrainer which is now deprecated (nnUNetTrainerV2 performs better). You will need different ranges of learning rate for that.
Best,
Fabian

**Comment by Joevaen** [comment]
Thanks，I will do some try.
Best,
Joevaen.

**Comment by Sayantan17** [comment]
> Thanks，I will do some try.
> Best,
> Joevaen.

@Joevaen did you analysed the loss functions? I do have some questions, if you can address them, in email, kindly let me know. Can you?
